includes/Core/Util/Uninstallation.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Uninstallation
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\Encrypted_Options;
+use Google\Site_Kit\Core\Authentication\Credentials;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Exception;
+
+/**
+ * Utility class for handling uninstallation of the plugin.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Uninstallation {
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since n.e.x.t
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Options object.
+	 *
+	 * @since n.e.x.t
+	 * @var Options
+	 */
+	private $options = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * This class and its logic must be instantiated early in the WordPress
+	 * bootstrap lifecycle because an uninstallation hook runs decoupled from
+	 * regular action hooks like 'init'.
```

Comment:
This is mostly still true, but may be slightly inaccurate as the uninstall is no longer based on a WP uninstall hook.